### DISCUSSION TITLE
What does that "/z/" mean in the shortened URL "https://godbolt.org/z/bqefKzzxW"?

### DISCUSSION THREAD

**Discussion by DanielLangr** [compiler-explorer/compiler-explorer#6524] [General]
Is a particular case/state on godbolt.org fully determined by the last part of the shortened URL (like "bqefKzzxW" for "https://godbolt.org/z/bqefKzzxW")? Or does that "/z/" also need to be there as well? What does that "/z/" stands for?

**Comment by AbrilRBS** [comment]
Hi, thanks for your question!
Internally, the short links are identified by the string after the `/z/` bit, but for the site to understand that you're loading a short link, the `/z/` needs to be in the URL, as that's the route that handles the loading of such links

Afaik the decision for the link to be a `/z/` is random, but probably @mattgodbolt knows better if he remembers :)

**Comment by AbrilRBS** [comment]
Oh, as per https://github.com/compiler-explorer/compiler-explorer/pull/1036 , I'm the one that added the route? Welp then I have absolutely no idea if it was ever meant to stand for anything :P

**Reply by mattgodbolt** [reply]
I think we had discussed it, I recall the z being like gzip, zcompress, zstd...there's a lot of z in compression.

We had a /g/ or something for the old Google short URLs and needed something different.

**Reply by DanielLangr** [reply]
Thanks for clarification, but if the links are identified by the string after the `/z/`, and there also exists (old) links with `/g/`, couldn't happen that both share the same last part? Or, is it guaranteed that every such part is unique independently of that `/*/` in front of it?

**Reply by partouf** [reply]
`/z/` is pretty much guaranteed to stay (we make the identifiers longer if we feel we might be running out of space)
